apps/server/src/modules/school/api/test/school.controller.api.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { HttpStatus, INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { cleanupCollections, TestApiClient } from '@shared/testing';
+import {
+	federalStateFactory,
+	schoolFactory,
+	schoolYearFactory,
+	UserAndAccountTestFactory,
+} from '@shared/testing/factory';
+import { ServerTestModule } from '@src/modules/server';
+
+describe('School Controller (API)', () => {
+	let app: INestApplication;
+	let em: EntityManager;
+	let testApiClient: TestApiClient;
+
+	beforeAll(async () => {
+		const moduleFixture = await Test.createTestingModule({
+			imports: [ServerTestModule],
+		}).compile();
+
+		app = moduleFixture.createNestApplication();
+		await app.init();
+		em = app.get(EntityManager);
+		testApiClient = new TestApiClient(app, 'school');
+	});
+
+	beforeEach(async () => {
+		await cleanupCollections(em);
+	});
+
+	afterAll(async () => {
+		await app.close();
+	});
+
+	describe('getSchool', () => {
+		describe('when no user is logged in', () => {
+			it('should return 401', async () => {
+				const someId = new ObjectId().toHexString();
+
+				const response = await testApiClient.get(`id/${someId}}`);
+
+				expect(response.status).toEqual(HttpStatus.UNAUTHORIZED);
+			});
+		});
+
+		describe('when id in params is not a mongo id', () => {
+			const setup = async () => {
+				const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent();
+
+				await em.persistAndFlush([studentAccount, studentUser]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(studentAccount);
+
+				return { loggedInClient };
+			};
+
+			it('should return 400', async () => {
+				const { loggedInClient } = await setup();
+
+				const response = await loggedInClient.get(`id/123`);
+
+				expect(response.status).toEqual(HttpStatus.BAD_REQUEST);
```

Comment:
expect(response.body) soll jeweils noch geprüft werden.

Reply:
I added a test for the validationErrors but I had to disable ESLint on this line because the response from the TestApiClient is not typed. 

In other files where this is tested the TestApiClient is not used, e.g. https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/modules/files-storage/controller/api-test/files-storage-list-files.api.spec.ts#L101

Did you envision something for this situation when you introduced the TestApiClient?

Reply:
Ich hab nen Test für die validationErrors an der Stelle hinzugefügt, aber musste ESLint in der Zeile deaktivieren, weil die Response vom TestApiClient nicht entsprechend typisiert ist.

An anderen Stellen, wo das getestet wird, wird der TestApiClient nicht benutzt, z.B.: https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/modules/files-storage/controller/api-test/files-storage-list-files.api.spec.ts#L101

In den anderen Tests mit dem TestApiClient seh ich keine Prüfung der validationErrors. Gibts nen Plan dafür?

Reply:
Ja schon. Der Plan war immer status code+body und bei Änderungen in der DB den change zu prüfen. 
Die Einhaltung der Vorgabe ist dann ein anderes Thema.

Ich denke der as cast bei der verlinkten Stelle ist auch nur scheinbar richtig. Aber auch nicht besser als die Ausnahme für das expect. Ich finde das erstmal ok mit der Ausnahme. Kann man sich mal anschauen wenn man viel Zeit hat. Also nicht jetzt und niemals sonst. :D

Reply:
Soll in jedem Fehlerfall der ganze Body abgetestet werden oder nur bei so Spezialfällen, z.B. die ValidationErrors? Gibts einen Test, wo das so ist, wie es sein soll?